### PR TITLE
build: Update traccc to v0.16.0

### PR DESCRIPTION
**Pull request by stephenswat**
This commit updates traccc to version 0.16.0 and, with it, detray to v0.75.2.

### CODE REVIEW

**Comment by asalzburger** [comment]
@stephenswat the version synchronisation went in, so this can be updated now.

**Comment by stephenswat** [comment]
@asalzburger updated.

**Comment by niermann999** [comment]
> ` #include <detray/test/utils/inspectors.hpp>`
> 
> has now moved to `test_utils` - right @niermann999 ?

yes, correct

File: thirdparty/detray/CMakeLists.txt [hunk 1]
```diff
@@ -30,11 +30,21 @@ set(DETRAY_CUSTOM_SCALARTYPE
...
+set(DETRAY_BUILD_UNITTESTS
     OFF
     CACHE BOOL
     "Turn off the build of the Detray unit tests"
 )
+set(DETRAY_BUILD_INTEGRATIONTESTS
+    OFF
+    CACHE BOOL
+    "Turn off the build of the Detray integration tests"
+)
```
Comment (niermann999) [inline thirdparty/detray/CMakeLists.txt:47]: I would also switch off the benchmarks here (```DETRAY_BUILD_BENCHMARKS```)

Comment (stephenswat) [reply thirdparty/detray/CMakeLists.txt:47]: Done.

**Review by niermann999 — DISMISSED** [review]

**Comment by asalzburger** [comment]
I still saw a weird build error associated with `actsvg::core` in the clang-tidy build. Rebased, and let's see.

**Comment by stephenswat** [comment]
I will look into this after the morning meeting.

**Comment by stephenswat** [comment]
Found the clang-tidy error, fixed in #3639.

**Comment by asalzburger** [comment]
Waiting for #3639 to be merged in.

**Review by asalzburger — APPROVED** [review]

**Comment by acts-project-service** [comment]
## :red_circle: Athena integration test results [f404ca2fa85ab545442f2d5680017a1205e39678]
### Build job with this PR failed!
Please investigate the [build job](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/43882932) for the [pipeline](https://gitlab.cern.ch/acts/acts-athena-ci/-/pipelines/8192686)!

**Pull request merged**